### PR TITLE
Remove passing compilation options to resolution in package compilation

### PR DESCRIPTION
**Pull request by pramodya1994**
- Fixes https://github.com/ballerina-platform/ballerina-lang/issues/34949

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageCompilation.java [hunk 1]
```diff
@@ -71,7 +71,7 @@ private PackageCompilation(PackageContext rootPackageContext) {
 
     private PackageCompilation(PackageContext rootPackageContext, CompilationOptions compilationOptions) {
         this.rootPackageContext = rootPackageContext;
-        this.packageResolution = rootPackageContext.getResolution(compilationOptions);
+        this.packageResolution = rootPackageContext.getResolution();
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageCompilation.java:74]: We can refactor this class to remove duplicated code now. Shall we do that improvment?

Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageCompilation.java:67]: Refactored via https://github.com/ballerina-platform/ballerina-lang/pull/35067/commits/f2736d713095205a010a22682fb483e0c3965d06

**Comment by azinneera** [comment]
Shall we add a CLI test for this?

**Comment by azinneera** [comment]
The PR doesn't fix the issue. A new one should be reopened once the proper fix is done .

**Pull request closed without merging**